tests/v3_api/common.py

```diff
@@ -232,6 +232,40 @@ def validate_workload_with_sidekicks(p_client, workload, type, ns_name,
         assert "running" in pod["status"]["containerStatuses"][1]["state"]
 
 
+def validate_workload_paused(p_client, workload, expectedstatus):
+    workloadStatus = p_client.list_workload(uuid=workload.uuid)[0].paused
+    assert workloadStatus == expectedstatus
+
+
+def validate_pod_images(expectedimage, workload, ns_name):
+    pods = execute_kubectl_cmd("get pods -l workload.user.cattle.io/"
+                               "workloadselector=deployment-{0}-{1}"
+                               .format(ns_name, workload.name))
+    for pod in pods["items"]:
+        assert pod["spec"]["containers"][0]["image"] == expectedimage
+
+
+def validate_pods_are_running_by_id(expectedpods, workload, ns_name):
+    allpods = execute_kubectl_cmd("get pods -l workload.user.cattle.io/"
+                                  "workloadselector=deployment-{0}-{1}"
+                                  .format(ns_name, workload.name))
+    for expectedpod in expectedpods["items"]:
+        expectedpodname = expectedpod["metadata"]["name"]
+        for currentpod in allpods["items"]:
+            if currentpod["metadata"]["name"] == expectedpodname:
+                assert currentpod["status"]["phase"] == "Running"
+
+
+def validate_workload_image(client, workload, expectedImage, ns):
+    workload = client.list_workload(uuid=workload.uuid)[0]
+    assert workload.containers[0].image == expectedImage
+    validate_pod_images(expectedImage, workload, ns.name)
+
+
+def validate_revision_count(revision, expectedLength):
+    assert len(revision) == expectedLength
+
```

Comment:
@adingilloRancher We dont need a separate method for this.  We can do the check directly for the length in places where it is needed.